apps/examples/stdk_smart_switch/device_control.c

```diff
@@ -0,0 +1,176 @@
+/* ***************************************************************************
+ *
+ * Copyright 2020 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <stdio.h>
+#include "device_control.h"
+#include <iotbus/iotbus_gpio.h>
+#include "iot_os_util.h"
+
+static iot_os_queue *button_event_queue = NULL;
+static iotbus_gpio_context_h g_pir;
+extern void iotapi_initialize(void);
+
+int gpio_set_level(int port, int value)
+{
+	//the gpio value need to refer the specific lamp device
+	value = (value == 0) ? NOTIFICATION_LED_GPIO_OFF : NOTIFICATION_LED_GPIO_ON;
+
+	iotbus_gpio_context_h ibctx = iotbus_gpio_open(port);
+	iotbus_gpio_write(ibctx, value);
+	iotbus_gpio_close(ibctx);
+	return OK;
+}
+
+static int gpio_get_level(int port)
+{
+	int result = 0;
+	iotbus_gpio_context_h ibctx = iotbus_gpio_open(port);
+	result = iotbus_gpio_read(ibctx);
+	iotbus_gpio_close(ibctx);
+	return result;
+}
+
+void button_isr_handler(void *arg)
+{
+	static uint32_t last_time_ms = 0;
+	uint32_t now_ms = TICK2MSEC(clock_systimer());
```

Comment:
headers for `uint32_t`, `TICK2MSEC` and `clock_systimer`

Reply:
fixed, and also for smart_lamp example, thanks.